### PR TITLE
Add discord messaging timestamps

### PR DESCRIPTION
**Pull request by warriorzz**
https://discord.com/developers/docs/reference#message-formatting

### CODE REVIEW

File: common/src/main/kotlin/DiscordTimestamp.kt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    override fun toString() = value
+}
+
+fun LocalDateTime.toDiscordTimestamp(style: DiscordTimestampStyle? = null) =
+    DiscordTimestamp(toInstant(TimeZone.UTC).toEpochMilliseconds(), style ?: DiscordTimestampStyle.ShortDateTime)
+
+fun Instant.toDiscordTimestamp(style: DiscordTimestampStyle? = null) =
+    DiscordTimestamp(toEpochMilliseconds(), style ?: DiscordTimestampStyle.ShortDateTime)
```
Comment (Lukellmann) [inline common/src/main/kotlin/DiscordTimestamp.kt:46]: I would make `LocalDateTime.toDiscordTimestamp` call `Instant.toDiscordTimestamp` but this is just a nit-pick.
And use `DiscordTimestampStyle.ShorrDateTime` as the default for `style`, no need for this to be nullable.

**Review by Lukellmann — CHANGES_REQUESTED** [review]

File: common/src/main/kotlin/DiscordTimestamp.kt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import kotlinx.datetime.Instant
+import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.toInstant
+import kotlinx.serialization.Serializable
+
+/**
+ * The [timestamp format](https://discord.com/developers/docs/reference#message-formatting-formats) of the message formats
+ */
+@Serializable
```
[the remark this answers is not in the document]
Comment (warriorzz) [reply common/src/main/kotlin/DiscordTimestamp.kt:12]: No real need, but I thought if anyone wants to insert this into their database, they don't need to create an extra data class

[the remark this answers is not in the document]
Comment (Lukellmann) [reply common/src/main/kotlin/DiscordTimestamp.kt:12]: Sure, did not think about that 😄

**Comment by BartArys** [comment]
What do we gain from introducing a DiscordTimestamp? Its sole reason for existence seems to be to convert it to a String. Why not just make it a String directly instead?

The entire lib uses kx's `Instant` for time already, it seems more prudent to introduce a `toMessageFormat` (or whatever you want to call it) extension that accepts your `DiscordTimestampStyle` to those types directly instead, and save the hassle of using an adapter type.

**Comment by warriorzz** [comment]
I didn't thought about that, but sounds good to me

File: common/src/main/kotlin/DiscordTimestamp.kt [hunk 3]
```diff
@@ -3,7 +3,7 @@ package dev.kord.common
 import kotlinx.datetime.Instant
 
 fun Instant.toMessageFormat(style: DiscordTimestampStyle = DiscordTimestampStyle.ShortDateTime) =
-    "<t:${toEpochMilliseconds()}:${style.style}>"
+    "<t:${toEpochMilliseconds() / 1000}:${style.style}>"
```
Comment (NyCodeGHG) [inline common/src/main/kotlin/DiscordTimestamp.kt:6]: better use epochSeconds instead?

Comment (NyCodeGHG) [reply common/src/main/kotlin/DiscordTimestamp.kt:6]:
```suggestion
    "<t:$epochSeconds:${style.style}>"
```
now i can suggest thanks github

**Pull request merged**